### PR TITLE
feat(openstack): glance v2 api in Openstack Clouddriver

### PR DESCRIPTION
**Pull request by posox**
Glance v1 api was deprecated in OpenStack Newton and removed in Openstack Rocky release.

https://developer.openstack.org/api-ref/image/versions/index.html#version-history

### CODE REVIEW

**Comment by lwander** [comment]
Could @varikin or someone with openstack experience take a look?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackImageProvider.groovy [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 package com.netflix.spinnaker.clouddriver.openstack.client
 
+import com.netflix.spinnaker.clouddriver.openstack.model.OpenstackImage
 import org.openstack4j.model.image.Image
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackImageProvider.groovy]: I think this import of `org.openstack4j.model.image.Image` is now unused for the interface.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackProviderFactory.groovy [hunk 2]
```diff
@@ -28,7 +28,7 @@ class OpenstackProviderFactory {
     OpenstackComputeV2Provider computeProvider = new OpenstackComputeV2Provider(identityProvider)
     OpenstackNetworkingProvider networkingProvider = new OpenstackNetworkingV2Provider(identityProvider)
     OpenstackOrchestrationProvider orchestrationProvider = new OpenstackOrchestrationV1Provider(identityProvider)
-    OpenstackImageProvider imageProvider = new OpenstackImageV1Provider(identityProvider)
+    OpenstackImageProvider imageProvider = new OpenstackImageV2Provider(identityProvider)
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackProviderFactory.groovy]: The `OpenstackImageV1Provider` is still in Clouddriver, but not being used. Please either have this factory select either v1 or v2 based on something or just remove the v1 provider so we don't have unused classes.

Comment (posox) [reply clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackProviderFactory.groovy]: Done

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackImageProvider.groovy (hunk 1 above)
Comment (posox) [reply clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackImageProvider.groovy]: Done

**Comment by lwander** [comment]
Looks good, seems the tests need to be addressed: https://travis-ci.org/spinnaker/clouddriver/builds/393609504#L1400

**Comment by varikin** [comment]
Needs to be rebased

**Comment by varikin** [comment]
For some reason, I am not getting notified when there is a new commit. I am sorry for not merging this earlier. If you rebase again and comment, I'll get it merged.

**Comment by varikin** [comment]
@lwander LTGM pls 👍

**Comment by msilpala** [comment]
Fun time. This change recently hit our prod environment, and it turns out not to be backward-compatible. Opening an issue to discuss the fix.

**Pull request merged**